packages/flutter/lib/src/widgets/icon_data.dart

```diff
@@ -99,6 +106,18 @@ class _StaticIconProvider {
 /// Annotation for classes that only provide static const [IconData] instances.
 ///
 /// This is a hint to the font tree shaker to ignore the constant instances
-/// of [IconData] appearing in the class when tracking which code points
-/// should be retained in the bundled font.
+/// of [IconData] appearing in the declaration of this class when tree-shaking
+/// unused code points from the bundled font.
+///
+/// Classes with this annotation should only have static const declarations of
+/// the [IconData] class, such as:
```

Comment:
nit: I would rephrase this, instead of saying "should" I would say, tree shaking only works if you do this. Otherwise your binary will be larger than necessary (or whatever the failure mode is).

Reply:
updated, WDYT?